hz.ps1

```diff
@@ -981,6 +1038,40 @@ if ($doTagVersion)
     }
 }
 
+# generate codecs
+if ($doCodecs) {
+
+    # note: codec files contain the client-side methods, along with the
+    # server-side methods enclosed in '#if SERVER_CODEC' blocks. files are
+    # copied to the client project, which uses them raw, and linked into
+    # the testing projects, which defines SERVER_CODEC.
+
+    # wipe existing codecs from the C# and protocol repositories
+    Write-Output "Clear codecs"
+    rm -force $srcDir/Hazelcast.Net/Protocol/Codecs/*.cs
+    rm -force $srcDir/Hazelcast.Net/Protocol/CustomCodecs/*.cs
+    rm -force -recurse $slnRoot/protocol/output/cs
+
+    Write-Output "Generate codecs"
+    python $slnRoot/protocol/generator.py -l cs --no-binary
+
+    # copy generated codecs to the C# repository
+    Write-Output "Copy codecs"
+    cp $slnRoot/protocol/output/cs/src/Hazelcast.Net/Protocol/Codecs/*.cs $srcDir/Hazelcast.Net/Protocol/Codecs/
+    cp $slnRoot/protocol/output/cs/src/Hazelcast.Net/Protocol/CustomCodecs/*.cs $srcDir/Hazelcast.Net/Protocol/CustomCodecs/
+
+    # normalize codecs FIXME platform?
+    Write-Output "Normalize codecs"
+    foreach ($file in $(ls $srcDir/Hazelcast.Net/Protocol/Codecs/*.cs)) {
+        clrSrc $file
```

Comment:
clrSrc function belongs to client-protocol repo's code generator.

Reply:
I get it